feat: add improve for Epd option, add active-color-label

flxzt opened the pull request:
This adds the following:

- A label that displays the current active color as a string in the format \<brightness\>,\<saturation\>,\<hue\> with easy to understand strings.
- An "optimize for E-Paper displays" option in order to toggle the visibility of this label. In the future more modifications to the app behaviour and UI additions can be enabled through it

Still a draft, because rnote-compose now needs to be translated with gettext as well. I'll see how to best integrate or refactor it

addresses #658

flxzt commented:
@Doublonmousse if you want/have enough time, please review!

Doublonmousse commented on crates/rnote-ui/po/rnote.pot:
```diff
@@ -2318,3 +2332,114 @@ msgstr ""
+
+#: crates/rnote-ui/src/utils.rs:161
+msgctxt "part of string representation of a color"
+msgid "desaturated"
+msgstr ""
+
+#: crates/rnote-ui/src/utils.rs:163
+msgctxt "part of string representation of a color"
+msgid "vibrant"
+msgstr ""
+
+#: crates/rnote-ui/src/utils.rs:167
+msgctxt "part of string representation of a color"
+msgid "dark"
+msgstr ""
+
+#: crates/rnote-ui/src/utils.rs:168
+msgctxt "part of string representation of a color"
+msgid "mid"
+msgstr ""
+
+#: crates/rnote-ui/src/utils.rs:169
+msgctxt "part of string representation of a color"
+msgid "bright"
+msgstr ""
+
+#: crates/rnote-ui/src/utils.rs:174
+msgctxt "part of string representation of a color"
+msgid "transparent"
+msgstr ""
+
+#: crates/rnote-ui/src/utils.rs:176
+msgctxt "part of string representation of a color"
+msgid "black"
+msgstr ""
+
+#: crates/rnote-ui/src/utils.rs:178
+msgctxt "part of string representation of a color"
+msgid "white"
+msgstr ""
```
The rule for `gettext` to work correctly is to match the line the `pgettext` is called in the `pot` file right ?
I'm not 100 % sure it's the case here

line number in `utils.rs` vs `rnote.pot` (and associated name)

148/145/rose
150/146/red
153/147/orange
156/148/yellow
158/149/chartreuse-green
163/150/green
166/151/spring-green
169/152/cyan
172/153/azure
175/154/blue
178/155/violet
181/156/magenta
183/157/rose
184/158/invalid

187/161/desaturated
189/163/vibrant
190/164/invalid

193/167/dark
195/168/mid
197/169/bright
198/170/invalid

202/174/transparent
204/176/black
206/178/white

flxzt replied:
I *think* the line number only helps for matching or even is only a help for translators to get additional context by looking at the source code, the msgid is what will be matched against in the end.
I don't know why it wrote the wrong line numbers, but it seems to be fixed with the latest execution of `meson compile rnote-pot` (though I only checked a few entries)

Doublonmousse reviewed (commented):
Alpha values are not shown in the name for the color ?

Testing with some colors, I do have some weird results on some slightly desaturated colors.

- #986A44, #63452C shows as mid orange 
- #FF7800 shows as white (it's a slightly desaturated orange)

I think it'd be good to plot where the limits are in the `Okhsv` space to double check this

flxzt commented:
Good points, thanks.

I think I misunderstood the `value` field in the hsv colorspace while implementing this. I now fixed and improved the logic by:
- combining checks for value **and** saturation to detect white and black
- accounting for float inaccuracies by using `approx::rel_eq` with f32's epsilon for min/max values. It's relevant because the picked colors come from gdk, which only has f32 precision.
- hand-tweaked the ranges for colors. This is the part which will probably need further testing, because I don't have a calibrated screen. But at least the Gnome colors in the colorpicker are now always in the right category. Ideally there would a definitive (scientific) source for defining hue values as strings, but didn't find one yet.
- added alpha strings `transparent`, `translucent`, `slightly translucent` and `` (empty string == opaque)
